### ISSUE TITLE
Allow custom size nodebox to have custom size textures

### ISSUE THREAD

**Issue by Yzelast** [minetest/minetest#7751]
##### Issue type
 - Feature request
##### Minetest version
```
"0.4.17.1"
```

##### OS / Hardware

Operating system: Elementary OS 0.4
CPU:Intel core i5 4300m

GPU model: Hd graphics 4600
OpenGL version: 3.0(according to minetest window)

##### Summary

Allow nodeboxes bigger than default node to have textures bigger too.

After testing some stuff with nodeboxes, i noticed that even that i can properly create nodeboxes with custom sizes, the textures only works right with 1 node sizes, so if i create an node with -1,1 size, the texture will be stretched to the center and then be repeated at the extra space.

In this example i have a vase with a 20x20x20 pixel size:
![captura de tela de 2018-09-26 17-03-04](https://user-images.githubusercontent.com/8459626/46109703-6bfd4c80-c1af-11e8-9935-c3028c6f6b0f.png)
Then i created a random texture with 20px size but noticed that the texture did not follow the box size.

In another test i have a default size 16x16x16:
![captura de tela de 2018-09-26 17-05-42](https://user-images.githubusercontent.com/8459626/46109900-ec23b200-c1af-11e8-810d-ad4de8b73b1d.png)
The 20px texture was reduced to 16px.

Basically what i request is to be able to create a custom nodebox shape, and have a texture that follow the same shape.
Ex: creating a nodebox with 32x16x32 and then having a texture with 32x16 that uses works with the custom size.

**Comment by tacotexmex** [comment]
An option for this is very much needed for me.

**Comment by paramat** [comment]
+1

**Comment by v-rob** [comment]
The only current alternative solution is making meshes in Blender.  That, of course, is not an ideal solution.

**Comment by runsy** [comment]
I need this feature too much.

[https://forum.minetest.net/viewtopic.php?f=47&t=25025](https://forum.minetest.net/viewtopic.php?f=47&t=25025)
